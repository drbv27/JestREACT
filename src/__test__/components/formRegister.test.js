import React from 'react'
import { create } from 'react-test-renderer'
import { mount, shallow } from 'enzyme'
import FormRegister from '../../components/formRegister'
import ProviderMock from '../../__mocks__/providerMock'

describe('<formRegister />', () => {
    test('Render del componente formRegister', () => {
        const formRegister = shallow (
        <ProviderMock>
        <FormRegister />
        </ProviderMock>
    )
        expect(formRegister.length).toEqual(1)
    })
})


describe('FormRegister Snapshot', () => {

    test('Comprobar la UI del componente FormRegister', ()=> {
        const formRegister = create(<FormRegister />)
        expect(formRegister.toJSON()).toMatchSnapshot()
    })

})
