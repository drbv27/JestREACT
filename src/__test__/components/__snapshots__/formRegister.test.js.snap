// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRegister Snapshot Comprobar la UI del componente FormRegister 1`] = `
<form
  className="user"
  onSubmit={[Function]}
>
  <div
    className="form-group row"
  >
    <div
      className="col-sm-6 mb-3 mb-sm-0"
    >
      <input
        className="form-control form-control-user"
        id="firstName"
        name="name"
        onChange={[Function]}
        placeholder="First Name"
        required={true}
        type="text"
      />
    </div>
    <div
      className="col-sm-6"
    >
      <input
        className="form-control form-control-user"
        id="lastName"
        name="lastName"
        onChange={[Function]}
        placeholder="Last Name"
        required={true}
        type="text"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <input
      className="form-control form-control-user"
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email Address"
      required={true}
      type="email"
    />
  </div>
  <div
    className="form-group row"
  >
    <div
      className="col-sm-6 mb-3 mb-sm-0"
    >
      <input
        className="form-control form-control-user"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="col-sm-6"
    >
      <input
        className="form-control form-control-user"
        id="confirmPassword"
        name="confirmPassword"
        onChange={[Function]}
        placeholder="Repeat Password"
        required={true}
        type="password"
      />
    </div>
  </div>
  <button
    className="btn btn-primary btn-user btn-block"
    type="submit"
  >
    Register Account
  </button>
</form>
`;
